File: app/src/main/java/soup/movie/di/ui/TheaterSortUiModule.java
package soup.movie.di.ui;

import dagger.Module;
import dagger.Provides;
import soup.movie.di.scope.ActivityScope;
import soup.movie.settings.TheaterSetting;
import soup.movie.ui.theater.sort.TheaterSortContract;
import soup.movie.ui.theater.sort.TheaterSortPresenter;

@Module
public class TheaterSortUiModule {

    @ActivityScope
    @Provides
    TheaterSortContract.Presenter presenter(TheaterSetting theaterSetting) {
        return new TheaterSortPresenter(theaterSetting);
    }
}

File: app/src/main/java/soup/movie/di/ui/TheaterSortUiModule.kt
package soup.movie.di.ui

import androidx.lifecycle.ViewModel
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector
import dagger.multibindings.IntoMap
import soup.movie.di.key.ViewModelKey
import soup.movie.di.scope.FragmentScope
import soup.movie.ui.theater.sort.TheaterSortFragment
import soup.movie.ui.theater.sort.TheaterSortViewModel

@Module
interface TheaterSortUiModule {

    @FragmentScope
    @ContributesAndroidInjector
    fun bindTheaterSortFragment(): TheaterSortFragment

    @Binds
    @IntoMap
    @ViewModelKey(TheaterSortViewModel::class)
    fun bindTheaterSortViewModel(viewModel: TheaterSortViewModel): ViewModel
}
